crates/sui-config/src/genesis_config.rs

```diff
@@ -110,14 +110,27 @@ pub struct ValidatorGenesisInfo {
     pub network_address: Multiaddr,
     pub p2p_address: Multiaddr,
     pub p2p_listen_address: Option<SocketAddr>,
+    #[serde(default = "default_socket_address")]
     pub metrics_address: SocketAddr,
+    #[serde(default = "default_multiaddr_address")]
     pub narwhal_metrics_address: Multiaddr,
     pub gas_price: u64,
     pub commission_rate: u64,
     pub narwhal_primary_address: Multiaddr,
     pub narwhal_worker_address: Multiaddr,
 }
 
+fn default_socket_address() -> SocketAddr {
+    utils::available_local_socket_address()
```

Comment:
Call `utils::available_local_socket_address()` is good for unit tests, but does this work for other deployments? Maybe we can test it out in private testnet?

Reply:
I will test it out in ci & private-testnet

Reply:
Tested with @andll and this seems to resolve the problems with private-testnet